rpc/legacyrpc/methods.go

```diff
@@ -3402,6 +3405,105 @@ WrongAddrKind:
 }
 
+// derive CointypeKey is to be used in verifySeed
+func deriveCoinTypeKey(Seed []byte, coinType uint32) (*hdkeychain.ExtendedKey, error) {
+
+	root, err := hdkeychain.NewMaster(Seed[:], &chaincfg.MainNetParams)
+	if err != nil {
+		return nil, err
+	}
+
+	// BIP0032 hierarchy: m/<purpose>'/
+	// Where purpose = 44 and the ` indicates hardening with the HardenedKeyStart 0x80000000
+	purpose, err := root.Child(44 + hdkeychain.HardenedKeyStart)
+	if err != nil {
+		return nil, err
+	}
+
+	// BIP0044 hierarchy: m/44'/<coin type>'
+	// Where coin type is either the legacy coin type, 20, or the coin type described in SLIP0044, 44.
+	inputtedCoinTypePrivKey, err := purpose.Child(coinType + hdkeychain.HardenedKeyStart)
+	if err != nil {
+		return nil, err
+	}
+
+	return inputtedCoinTypePrivKey, err
+}
+
+func verifySeed(s *Server, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*dcrjson.VerifySeedCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	// Changed inputted seed, type string, to type byte[] so hdkeychain methods can be utilized
+	Seed, err := walletseed.DecodeUserInput(cmd.Seed)
+	if err != nil {
+		return nil, err
+	}
+
+	coinType, err := w.GrabCoinType()
+	if err != nil {
+		return nil, err
+	}
+
+	inputtedCoinTypePrivKey, err := deriveCoinTypeKey(Seed, coinType)
+	if err != nil {
+		return nil, err
+	}
+
+	// Grab coin type private key from wallet for future comparison to the derived inputted coin type private key
+	walletCoinTypePrivKey, err := w.GrabCoinTypeKey()
+	if err != nil {
+		return nil, err
+	}
+
+	var result bool
+	switch {
+	case cmd.Account != nil:
+		// Both inputtedDerivedAccountKey and walletDerivedAccountKey use the BIP044 hierachy: m/44'/<coin type>'/<account>'
+		// If the child is a private extended key it is neutered
+		inputtedAccountKey, err := inputtedCoinTypePrivKey.Child(*cmd.Account + hdkeychain.HardenedKeyStart)
+		if err != nil {
+			return nil, err
+		}
+
+		inputtedxPubKey, err := inputtedAccountKey.Neuter()
+		if err != nil {
+			return nil, err
+		}
+
+		walletAccountKey, err := walletCoinTypePrivKey.Child(*cmd.Account + hdkeychain.HardenedKeyStart)
+		if err != nil {
+			return nil, err
+		}
+
+		walletxPubKey, err := walletAccountKey.Neuter()
+		if err != nil {
+			return nil, err
+		}
+
+		// Since the field, key, within the type struct, ExtendedKey, is private - keys must be converted
+		// to type string for comparison.
+		stringInputtedxPubKey := inputtedxPubKey.String()
+		stringWalletxPubKey := walletxPubKey.String()
+		result = strings.EqualFold(stringInputtedxPubKey, stringWalletxPubKey)
+
+	default:
+		stringInputtedCoinTypePrivKey := inputtedCoinTypePrivKey.String()
+
+		stringWalletCoinTypePrivKey := walletCoinTypePrivKey.String()
+
+		result = strings.EqualFold(stringInputtedCoinTypePrivKey, stringWalletCoinTypePrivKey)
+	}
+
+	return &dcrjson.VerifySeedResult{
+		Result:   result,
```

Comment:
Would prefer this field to be named "equal" or "matches" instead of the more ambiguous "result".